File: frontend/ubre/src/app/dtos/user-stats-dto.ts
export class UserStatsDto {
    public userId : number;
    public activePast24Hours : number;
    public numberOfRides : number;
    public distanceTravelled : number; 
    public moneySpent : number;
    public moneyEarned : number;

    constructor(userId: number, activePast24Hours : number, numberOfRides : number, distanceTravelled : number, moneySpent : number, moneyEarned : number) {
        this.userId = userId;
        this.activePast24Hours = activePast24Hours;
        this.numberOfRides = numberOfRides;
        this.distanceTravelled = distanceTravelled;
        this.moneySpent = moneySpent;
        this.moneyEarned = moneyEarned;
    }
}

File: backend/src/main/java/com/ubre/backend/dto/UserStatsDto.java
package com.ubre.backend.dto;

// User statistics object to be displayed in user profile (only necessary attributes for now are extracted)

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class UserStatsDto { // extend later with other attributes
    private Long userId;
    private Integer activePast24Hours; // should be in minutes
    private Integer numberOfRides;
    private Double distanceTraveled;
    private Double moneySpent;
    private Double moneyEarned;

    public UserStatsDto(Long userId, int activePast24Hours, int numberOfRides, double distanceTraveled, double moneySpent, double moneyEarned) {
        this.userId = userId;
        this.activePast24Hours = activePast24Hours;
        this.numberOfRides = numberOfRides;
        this.distanceTraveled = distanceTraveled;
        this.moneySpent = moneySpent;
        this.moneyEarned = moneyEarned;
    }
}
